api.yaml

```diff
@@ -15,6 +15,130 @@ tags:
     description: Execute, coordinate, or schedule operations
 
 paths:
+  /api/v1/threadpools:
```

Comment:
Is there a way we could auto-generate this spec. In the past I've used Django REST Framework which autogenerated the docs and schemas - https://www.django-rest-framework.org/topics/documenting-your-api/ . Quick search turned up this: https://github.com/outofcoffee/vertx-oas The reason we should consider doing this is, unless we're very diligent about keeping schemas in sync with the actual routes, the spec gets stale and goes out of sync from the APIs. Alternatively, we should maybe have a test that will check whether the schema is in conformance with the APIs. WDYT?

Reply:
👍 could be really nice to generate documentation from code using vertex-oas or something based on swagger-core.  One benefit of doing it by hand is you can come up with nice examples like what is done here, although maybe that is something you can express from code as well somehow.